[LISKDOCS-19] Introduction to Web3 app development

Tschakki opened the pull request.

ishantiw commented on user-guide.md:
````diff
@@ -62,53 +37,60 @@ COMMANDS
+```bash
+lisk config:set api.network <NETWORK> 
+```
+where `<NETWORK>` may be `main` for Mainnet and `test` for Testnet.
````
Could be improved,
Possible values for `<NETWORK>` are, `test` for `testnet, `main` for `mainnet`, `alpha` for `alphanet` and  `beta` for `betanet`

Tschakki committed:
Add pages

Tschakki committed:
Add PWA intro

Tschakki committed:
Add web3 app dev steps

Tschakki committed:
Fix conflicts

Tschakki committed:
Describe web versions

Tschakki committed:
Add Web 3 description

Tschakki committed:
Amend  Web 3 description

Tschakki committed:
Add section about getting started with Web3 development

Tschakki committed:
Compare Web2 and Web3 development

Tschakki committed:
Fix table

Tschakki committed:
Add descriptions

Tschakki committed:
Update structure

Tschakki committed:
Align headings

Tschakki committed:
Amend description

Tschakki committed:
Merge branch 'development' into liskdocs-19-mobile-dapp-guide

Tschakki committed:
Update examples

Tschakki committed:
Add examples

Tschakki committed:
Add examples

Tschakki committed:
Update descriptions

Tschakki committed:
Merge branch 'development' into liskdocs-19-mobile-dapp-guide

Tschakki committed:
Update resources

Tschakki committed:
Merge branch 'liskdocs-19-mobile-dapp-guide' of github.com:LiskHQ/lisk-documentation into liskdocs-19-mobile-dapp-guide

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+### Evolution of the Web
+
+Since the internet was born, there has been constant innovation in how people use the internet to communicate and interact with each other.
+To describe important steps in this innovation, websites and apps are generally categorized into different Web "versions":
```
```suggestion
To describe important steps in this innovation, websites, and apps are generally categorized into different Web "versions":
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+
+<Tabs queryString="web-version">
+  <TabItem value="web1" label="Web 1.0">
+    Web 1.0 is the first "version" of the Web, i.e. the first generation of websites that occured when the internet became available to the broader public around 1990.
```
```suggestion
    Web 1.0 is the first "version" of the Web, i.e., the first generation of websites that occurred when the Internet became available to the broader public around 1990.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+    Web 1.0 is the first "version" of the Web, i.e. the first generation of websites that occured when the internet became available to the broader public around 1990.
+    Web 1.0 is often referred to as the "static" or "read-only" web, consisting mainly of simple static HTML pages with limited interactivity.
+
+    However, some Web 2.0 capabilities were present in the days of Web 1.0, but were implemented differently.
```
```suggestion
    However, some Web 2.0 capabilities were present in the days of Web 1.0 but were implemented differently.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+
+### Web2 vs Web3 app development
+#### Languages
+Concerning frontend app development, there aren't many differences between Web2 and Web3 apps, when it comes to programming languages.
```
```suggestion
Concerning frontend app development, there aren't many differences between Web2 and Web3 apps when it comes to programming languages.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+Theoretically, you could build a Web3 app with nearly the same tech stack as a Web2 app, like Next.js, Vite, etc.
+The only thing you need to add is a way for your application to interact with the blockchain, see [Data access](#data-access).
+
+To support developers, many blockcain platforms offer their own SDKs and tools to interact with their blockchain, which can make development easier and more efficient.
```
```suggestion
To support developers, many blockchain platforms offer their own SDKs and tools to interact with their blockchain, which can make development easier and more efficient.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+
+#### Data storage
+In Web2 apps, data is typically stored in a centralized database.
+In Web3 apps, some data is stored on the blockchain, i.e. a decentralized database.
```
```suggestion
In Web3 apps, some data is stored on the blockchain, i.e., a decentralized database.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+#### Data storage
+In Web2 apps, data is typically stored in a centralized database.
+In Web3 apps, some data is stored on the blockchain, i.e. a decentralized database.
+How the data is stored is defined inside of smart cotracts.
```
```suggestion
How the data is stored is defined inside of smart contracts.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+In Web2 apps, data from the database is accessed through APIs.
+In Web3 apps, blockchain data is accessed through smart contracts.
+They usually expose public functions that can be called by the app frontend to read and/or write data from/to the blockchain, similar to a classical API.
+To interact with the blockchain, there are various libraries and frameworks available, like [viem](https://viem.sh/), [web3.js](https://web3js.readthedocs.io) or [ethers.js](https://docs.ethers.io).
```
```suggestion
To interact with the blockchain, there are various libraries and frameworks available, like [viem](https://viem.sh/), [web3.js](https://web3js.readthedocs.io), or [ethers.js](https://docs.ethers.io).
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+
+## Getting started with Web3 app development
+
+To get started with the actual development of a Web3 app, you need to be clear about the the right app **platform(s)** to build your Web3 app on.
```
```suggestion
To get started with the actual development of a Web3 app, you need to be clear about the right **platform(s)** to build your Web3 app on.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+
+### Web3 Web Apps
+The main advantage of web applications is that they are platform-independent and can be accessed from any device with a web browser.
+Additionally, they don't need to be installed by the user, and can be used directly in the browser.
```
```suggestion
Additionally, they don't need to be installed by the user and can be used directly in the browser.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+- [How to Build an NFT Minting dApp](https://blog.thirdweb.com/guides/build-nft-minting-dapp-with-thirdweb-guide/) *by thirdweb*
+
+### Web3 PWAs
+A Progressive Web App (= PWA) allows users to install your web app to their device.
```
```suggestion
A Progressive Web App (PWA) allows users to install a web app on their devices.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+This way, PWAs allow to develop an application that can be used via browser[^1], mobile, or desktop application alike.
+Therefore, PWAs are a great choice to reach users on different platforms simultaneously.
+
+Additionally, PWAs allow a user to directly install the app through your web app, without need to visit an app store.
```
```suggestion
Additionally, PWAs allow a user to directly install the app through your web app without the need to visit an app store.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+This allows you to serve the app directly to the users, without bothering about restrictions of app stores on what is allowed for blockchain applications.
+In-app purchase fees can also be avoided because users no longer need to go through the app store.
+
+[^1]: Only supported in PWA-compatible browsers (most Chromium based browsers should work).
```
```suggestion
[^1]: Only supported in PWA-compatible browsers (most Chromium-based browsers should work).
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+    The term "Web 2.0" was coined by Darcy DiNucci, an information architecture consultant, in her January 1999 article [Fragmented Future](http://darcyd.com/fragmented_future.pdf).
+    :::
+
+    Web 2.0 is the era where social media, [web applications](https://en.wikipedia.org/wiki/Web_application), and mobile apps come to the rise.
```
```suggestion
    Web 2.0 is the era where social media, [web applications](https://en.wikipedia.org/wiki/Web_application), and mobile apps came to the rise.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+
+    Web 2.0 is the era where social media, [web applications](https://en.wikipedia.org/wiki/Web_application), and mobile apps come to the rise.
+    Because of the new social features and improved user experience that Web 2.0 apps offered, the number of users interacting with the web increased significantly, leading to the creation of many new web services and applications.
+    Some of the most popular examples for Web 2.0 applications are Facebook, Twitter/X, Amazon and YouTube.
```
```suggestion
    Some of the most popular examples of Web 2.0 applications are Facebook, X, Amazon, and YouTube.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+    Because of the new social features and improved user experience that Web 2.0 apps offered, the number of users interacting with the web increased significantly, leading to the creation of many new web services and applications.
+    Some of the most popular examples for Web 2.0 applications are Facebook, Twitter/X, Amazon and YouTube.
+
+    Many of these applications and services became important pillars of the internet and peoples life, and even society as a whole.
```
```suggestion
    Many of these applications and services became important pillars of the internet and people's lives, and even society as a whole.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+    Some of the most popular examples for Web 2.0 applications are Facebook, Twitter/X, Amazon and YouTube.
+
+    Many of these applications and services became important pillars of the internet and peoples life, and even society as a whole.
+    However, the more relevant these applications became, the more obvious and relevant became their limitations, as users had to trust the companies behind these applications with their data and privacy.
```
```suggestion
    However, the more relevant these applications became, the more obvious and relevant their limitations became, as users had to trust the companies behind them with their data and privacy.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+
+    <h5>Web2 limitations</h5>
+
+    Though Web 2.0 revolutionized how people use the web in a positive way, it's characteristics lead to certain disadvantages, for users who interact with Web2 applications:
```
```suggestion
    Though Web 2.0 revolutionized how people positively used the web, a few of its characteristics led to certain disadvantages for users interacting with Web2 applications. For example:
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+  </TabItem>
+  <TabItem value="web3" label="Web3" default>
+    Web3 (also known as Web 3.0) is the next step in the evolution of the Web, aiming to address the [limitations of Web 2.0](web3-app-development.mdx?web-version=web2#web2-limitations).
+    Web3 is the idea of a decentralized internet, where users are in control over their data.
```
```suggestion
    Web3 is based on the idea of a decentralized internet, where users are in control over their data.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+    The term "Web3" was coined by Polkadot founder and Ethereum co-founder Gavin Wood in 2014, referring to a [decentralized online ecosystem based on the blockchain](https://www.wired.com/story/web3-gavin-wood-interview/).
+    :::
+
+    2015, when Ethereum launched as the world's first programmable blockchain, marked the beginning of the Web3 era.
```
```suggestion
    2015 marked the beginning of the Web3 era when Ethereum launched as the world's first programmable blockchain.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+
+    Web3 apps are built on top of blockchain technology, which allows users greater ownership and control over their data.
+    This provides new features and allows services that are not feasible in the Web2 world.
+    Often, Web3 apps combine other cutting edge technologies like AI to provide unique new user experiences.
```
```suggestion
    Often, Web3 apps combine other cutting-edge technologies like AI to provide unique and new user experiences.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+      - DEXs (Decentralized Exchanges)
+      - GameFi (Gaming Finance)
+      - Staking and yield farming platforms
+      - Decentralized crowdfunding and -investing platforms
```
```suggestion
      - Decentralized crowdfunding and investing platforms
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+    - [DeSci](https://onchain.org/magazine/not-only-the-best-desci-crypto-projects-an-overview/) (Decentralized Science)
+    - [DePIN](https://onchain.org/magazine/how-depin-web3-projects-are-connecting-the-world-to-crypto/) (Decentralized Physical Infrastructure Networks)
+  
+    Web3 is a promise to bring back the original vision of the internet, where users are in control of their data and privacy, and where the internet is a place for free speech and innovation.
```
```suggestion
    Web3 promises to bring back the original vision of the internet, where users are in control of their data and privacy and where the internet is a place for free speech and innovation.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+
+#### Development resources
+- [How to Create a Web3 PWA](https://blog.thirdweb.com/guides/how-to-create-a-web3-pwa/) *by thirdweb*
+- [Video tutorial: Build a web3 PWA](https://www.youtube.com/watch?v=VU8i-dn2_GE) *by thirdweb*
```
```suggestion
- [Video tutorial: Build a Web3 PWA](https://www.youtube.com/watch?v=VU8i-dn2_GE) *by thirdweb*
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+
+### Web3 PWAs
+A Progressive Web App (= PWA) allows users to install your web app to their device.
+They appear as native apps on your desktop or mobile device and offer features like offline support and push notifications.
```
```suggestion
They appear as native apps on desktop or mobile devices and offer features like offline support and push notifications.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+### Web3 PWAs
+A Progressive Web App (= PWA) allows users to install your web app to their device.
+They appear as native apps on your desktop or mobile device and offer features like offline support and push notifications.
+This way, PWAs allow to develop an application that can be used via browser[^1], mobile, or desktop application alike.
```
```suggestion
This way, PWAs allow the development of an application that can be used via a browser[^1], mobile, or desktop application alike.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,184 @@
+A Progressive Web App (= PWA) allows users to install your web app to their device.
+They appear as native apps on your desktop or mobile device and offer features like offline support and push notifications.
+This way, PWAs allow to develop an application that can be used via browser[^1], mobile, or desktop application alike.
+Therefore, PWAs are a great choice to reach users on different platforms simultaneously.
```
```suggestion
Therefore, PWAs are a great choice for reaching users on different platforms simultaneously.
```

TalhaMaliktz reviewed (changes requested):
Please consider the following!

Tschakki committed:
Apply suggestions from code review

Co-authored-by: Muhammad Talha <13951043+TalhaMaliktz@users.noreply.github.com>

Tschakki committed:
Merge branch 'development' into liskdocs-19-mobile-dapp-guide

TalhaMaliktz reviewed (approved):
LGTM 👍🏼

TalhaMaliktz reviewed (changes requested):
Broken anchor found!
<img width="1018" alt="image" src="https://github.com/user-attachments/assets/d00fe16c-2553-438e-8794-5ed59d429cf6" />

Tschakki commented:
> Broken anchor found! <img alt="image" width="1018" src="https://private-user-images.githubusercontent.com/13951043/401963303-d00fe16c-2553-438e-8794-5ed59d429cf6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY1MDc4NjIsIm5iZiI6MTczNjUwNzU2MiwicGF0aCI6Ii8xMzk1MTA0My80MDE5NjMzMDMtZDAwZmUxNmMtMjU1My00MzhlLTg3OTQtNWVkNTlkNDI5Y2Y2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEwVDExMTI0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3MGQ5NTczYTFjMWE5NTI0YTI2NzU2MTgxOWJiM2I0ODk1ODAwNmMzYjYyMWE1MTM4ZWM1Y2Q4YzAzODdhODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.pC_wQI27b2WsBF8fB3kyrFEqzVIS0qu6oSADZgji374">

this seems to be in the translation `/ind/`, which doesn't exist yet for this content... How did you get the error?

TalhaMaliktz commented:
> > Broken anchor found! <img alt="image" width="1018" src="https://private-user-images.githubusercontent.com/13951043/401963303-d00fe16c-2553-438e-8794-5ed59d429cf6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY1MDc4NjIsIm5iZiI6MTczNjUwNzU2MiwicGF0aCI6Ii8xMzk1MTA0My80MDE5NjMzMDMtZDAwZmUxNmMtMjU1My00MzhlLTg3OTQtNWVkNTlkNDI5Y2Y2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEwVDExMTI0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3MGQ5NTczYTFjMWE5NTI0YTI2NzU2MTgxOWJiM2I0ODk1ODAwNmMzYjYyMWE1MTM4ZWM1Y2Q4YzAzODdhODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.pC_wQI27b2WsBF8fB3kyrFEqzVIS0qu6oSADZgji374">
> 
> this seems to be in the translation `/ind/`, which doesn't exist yet for this content... How did you get the error?

I got it when I ran `npm run build && npm run serve`; if it is missing a file, I think we can make it part of our workflow to add a file in the `/ind` folder as well whenever we have the final and approved English version of the new file.

matjazv reviewed (approved):
Current documentation looks good to me!

Maybe consider extending it with below topics if you find them useful for this page:
- Security considerations in Web3 applications (smart contract audits, secure key management, handling user data, bug bounties)
- Testing and deployment (local development env, testnets, continuous integration, monitoring)
- Onboarding Web2 users (abstracting wallet setup, gas fee management, support and community)

Tschakki committed:
Fix broken anchor

Tschakki commented:
> > > Broken anchor found! <img alt="image" width="1018" src="https://private-user-images.githubusercontent.com/13951043/401963303-d00fe16c-2553-438e-8794-5ed59d429cf6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY1MDc4NjIsIm5iZiI6MTczNjUwNzU2MiwicGF0aCI6Ii8xMzk1MTA0My80MDE5NjMzMDMtZDAwZmUxNmMtMjU1My00MzhlLTg3OTQtNWVkNTlkNDI5Y2Y2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEwVDExMTI0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3MGQ5NTczYTFjMWE5NTI0YTI2NzU2MTgxOWJiM2I0ODk1ODAwNmMzYjYyMWE1MTM4ZWM1Y2Q4YzAzODdhODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.pC_wQI27b2WsBF8fB3kyrFEqzVIS0qu6oSADZgji374">
> > 
> > 
> > this seems to be in the translation `/ind/`, which doesn't exist yet for this content... How did you get the error?
> 
> I got it when I ran `npm run build && npm run serve`; if it is missing a file, I think we can make it part of our workflow to add a file in the `/ind` folder as well whenever we have the final and approved English version of the new file.

I added the file, which did not resolve the error. Then I changed format of the heading, and it fixed it, so I changed it in the original file as well.. not sure why the error was showing up for the translation and not original file though 🤔

TalhaMaliktz reviewed (approved):
LGTM! 👍🏼

Tschakki committed:
Add new sections

Tschakki committed:
Summarize account abstraction

Tschakki committed:
Add test examples

Tschakki committed:
Merge branch 'development' into liskdocs-19-mobile-dapp-guide

Tschakki committed:
Align headings

Tschakki committed:
Add links

Tschakki committed:
Add guides

Tschakki committed:
Move section

Tschakki committed:
Fix links

Tschakki committed:
Move section

Tschakki committed:
Fix link

Tschakki committed:
Describe testnets

Tschakki committed:
Describe e2e and monitoring

Tschakki committed:
Clean up text

Tschakki committed:
Update ToC levels

Tschakki committed:
Fix liskdocs-43, update translations

Tschakki committed:
liskdocs-45

Tschakki committed:
Update sidebars

Tschakki committed:
Update sidebars

matjazv commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,335 @@
+There are also SDKs for Web3 app development that provide a complete tech stack, like [thirdweb](https://thirdweb.com/) or [Alchemy](https://www.alchemy.com/), which take away most of the complexity of blockchain development and provide a smooth developer experience.
+thirdweb supports [Lisk](https://thirdweb.com/lisk) and many other EVM-compatible blockchains.
+
+#### Data Storage
```
Perhaps add a few words regarding data privacy and visibility on-chain.
Any data on a public blockchain is also viewable by anyone. Developers need to be aware that:
- Once data is added to a public blockchain, it becomes virtually impossible to remove and is visible to everyone.
- Sensitive or personal data may require additional measures (for example, encryption or zero-knowledge techniques) to avoid unwanted exposure.

Tschakki committed:
Mention that sensitive onchain data should be encrypted

matjazv commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -325,7 +325,9 @@ thirdweb supports [Lisk](https://thirdweb.com/lisk) and many other EVM-compatibl
-However, not all data should always be stored on the blockchain, as it is quite expensive and slow to store large amounts of data onchain.
+However, it should be noted that onchain data on a public blockchain is visible to everyone and therefore not suitable for all types of data.
+Especially sensistive and private user data should not be stored on the blockchain, except it is encrypted.
+In addition, soring large amounts of data onchain is quite expensive and slow to store large amounts of data onchain.
```
```suggestion
In addition, storing large amounts of data onchain is both expensive and slow.
```

Tschakki committed:
Update docs/building-on-lisk/web3-app-development.mdx

Co-authored-by: Matjaz Verbole <verbole@gmail.com>

matjazv reviewed (approved):
LGTM!
Nice work!

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -100,66 +100,36 @@ To describe important steps in this innovation, websites, and apps are generally
-To interact with the blockchain, there are various libraries and frameworks available, like [viem](https://viem.sh/), [web3.js](https://web3js.readthedocs.io), or [ethers.js](https://docs.ethers.io).
-
-### Further regarding
+### Further Regarding
```
```suggestion
### Further Reading
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,337 @@
+
+### Onboarding Web2 Users to Web3
+
+A major challenge of Web3 applications is to provide a seamless user experience for users, similar to what they are used to from Web2 applications.
```
```suggestion
A significant challenge for Web3 applications is delivering a user experience that is as seamless as what users expect from Web2 applications.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,337 @@
+
+A major challenge of Web3 applications is to provide a seamless user experience for users, similar to what they are used to from Web2 applications.
+
+To onboard Web2 users to Web3, apps need to simplify the way how users interact with the app, in a way that feels natural and intuitive to them, while still providing the benefits of Web3 in a secure way.
```
```suggestion
To onboard Web2 users to Web3, apps need to simplify user interaction so that it feels natural and intuitive to them while still providing the benefits of Web3 in a secure way.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,337 @@
+A major challenge of Web3 applications is to provide a seamless user experience for users, similar to what they are used to from Web2 applications.
+
+To onboard Web2 users to Web3, apps need to simplify the way how users interact with the app, in a way that feels natural and intuitive to them, while still providing the benefits of Web3 in a secure way.
+To aim for a smooth user experience, and to abstract the complexity of blockchain technology away from the user, it is highly recommended to include "account abstraction" (sometimes also referred to as "smart accounts", "smart wallets" or "smart contract accounts") into your Web3 application.
```
```suggestion
To aim for a smooth user experience, and to abstract the complexity of blockchain technology away from the user, it is highly recommended to include [Account Abstraction](https://www.alchemy.com/overviews/what-is-account-abstraction) (sometimes also referred to as "smart accounts", "smart wallets" or "smart contract accounts") into your Web3 application.
```

Tschakki replied:
I wouldn't add a link to an external site here, as we explain below account abstraction ourselves to the reader. We could add it under the further reading section, if you consider this resource to be valuable.

TalhaMaliktz replied:
Sure, the further reading section sounds suitable as well! I would use this [Blog](https://www.alchemy.com/blog/account-abstraction), which I found very useful for understanding AA.

Tschakki replied:
added the link below!

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,337 @@
+
+It comprises the following fundamental components:
+
+- **UserOperation**: a structure that describes a transaction to be sent on behalf of a user.
```
```suggestion
- **UserOperation**: A structure that describes a transaction to be sent on behalf of a user.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,337 @@
+
+- **UserOperation**: a structure that describes a transaction to be sent on behalf of a user.
+To avoid confusion, it is not named “transaction”.
+Like a transaction, it contains “sender”, “to”, “calldata”, “maxFeePerGas”, “maxPriorityFee”, “signature”, “nonce”.
```
```suggestion
Like a transaction, it contains `sender`, `to`, `calldata`, `maxFeePerGas`, `maxPriorityFee`, `signature`, `nonce`.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,337 @@
+- **UserOperation**: a structure that describes a transaction to be sent on behalf of a user.
+To avoid confusion, it is not named “transaction”.
+Like a transaction, it contains “sender”, “to”, “calldata”, “maxFeePerGas”, “maxPriorityFee”, “signature”, “nonce”.
+Unlike a transaction the “signature” field usage is not defined by the protocol, but by each account implementation.
```
```suggestion
Unlike a transaction, the use of the `signature` field is not defined by the protocol but by each account implementation.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,337 @@
+Like a transaction, it contains “sender”, “to”, “calldata”, “maxFeePerGas”, “maxPriorityFee”, “signature”, “nonce”.
+Unlike a transaction the “signature” field usage is not defined by the protocol, but by each account implementation.
+Also, it contains several other fields, described in detail in ERC-4337.
+- **Account Contract**: the user's smart account sending a user operation.
```
```suggestion
- **Account Contract**: The user's smart account sending a user operation.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -0,0 +1,337 @@
+Unlike a transaction the “signature” field usage is not defined by the protocol, but by each account implementation.
+Also, it contains several other fields, described in detail in ERC-4337.
+- **Account Contract**: the user's smart account sending a user operation.
+- **EntryPoint**: a singleton contract to execute bundles of UserOperations.
```
```suggestion
- **EntryPoint**: A singleton contract to validate and execute bundles of UserOperations.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+Tests are written in JavaScript / Typescript.
+
+##### Guides
+- [Deploying a smart contract with Foundry](/building-on-lisk/deploying-smart-contract/with-Foundry.md)
```
```suggestion
- [Deploying a smart contract with Foundry](/building-on-lisk/deploying-smart-contract/with-Foundry#testing-the-smart-contract)
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+
+##### Guides
+- [Deploying a smart contract with Foundry](/building-on-lisk/deploying-smart-contract/with-Foundry.md)
+- [Deploying a smart contract with Hardhat](/building-on-lisk/deploying-smart-contract/with-Hardhat.md)
+- [Deploying a smart contract with Remix](/building-on-lisk/deploying-smart-contract/with-Remix.mdx)
```
The links should be replaced with guides about testing with Hardhat and Remix.

Tschakki replied:
Yes, we should extend the guides with examples how to test the smart contract, I created the corresponding issues now 👍

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+- [Slither](https://github.com/crytic/slither): Slither is a Solidity static analysis framework that detects common vulnerabilities in smart contracts.
+- [Etheno](https://github.com/crytic/etheno): Etheno is a tool that helps developers write secure smart contracts by providing a set of rules that can be used to check the security of smart contracts.
+- [Mythril](https://github.com/Consensys/mythril): Mythril is a security analysis tool for Ethereum smart contracts.
+- **Smart contract audits**:
```
Should this be removed?
```suggestion
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+- **Get an unlimited faucet** and stop begging for test ETH.
+Mint both native and ERC-20 tokens in any amount you need, impersonate accounts, and test complex scenarios uninterruptedly.
+- **Fully customize your development infrastructure** to fit your unique workflows.
+Virtual TestNets are easy to set up and configure, giving you full control and flexibility over your environments.
```
```suggestion
Virtual Testnets are easy to set up and configure, giving you full control and flexibility over your environments.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+#### E2E Testing
+End-to-end (E2E) testing is a type of software testing that tests the complete flow of an application from start to finish.
+This specifically includes testing the frontend of your application, ensuring that the user interface works as expected.
+To deal with stuff like wallets, and other Web3-specific things, you might want to use specialized Web3 E2E testing tools:
```
```suggestion
To deal with stuff like wallets, and other Web3-specific things, you might want to use specialized Web3 E2E testing tools.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+This specifically includes testing the frontend of your application, ensuring that the user interface works as expected.
+To deal with stuff like wallets, and other Web3-specific things, you might want to use specialized Web3 E2E testing tools:
+
+[Synpress](https://synpress.io/) is an E2E (End-to-End) testing library for Web3 dapps. 
```
Already abbreviated
```suggestion
[Synpress](https://synpress.io/) is an E2E testing library for Web3 dapps. 
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+To deal with stuff like wallets, and other Web3-specific things, you might want to use specialized Web3 E2E testing tools:
+
+[Synpress](https://synpress.io/) is an E2E (End-to-End) testing library for Web3 dapps. 
+Synpress can be used with two leading E2E testing frameworks out there, i.e., both [Playwright](https://playwright.dev/) and [Cypress](https://www.cypress.io/), with support for all of their its features.
```
```suggestion
Synpress can be used with two leading E2E testing frameworks out there, i.e., [Playwright](https://playwright.dev/) and [Cypress](https://www.cypress.io/), with support for all of their features.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+The only thing you need to add is a way for your application to interact with the blockchain, see [Data access](#data-access).
+
+To support developers, many blockchain platforms offer their own SDKs and tools to interact with their blockchain, which can make development easier and more efficient.
+There are also SDKs for Web3 app development that provide a complete tech stack, like [thirdweb](https://thirdweb.com/) or [Alchemy](https://www.alchemy.com/), which take away most of the complexity of blockchain development and provide a smooth developer experience.
```
```suggestion
There are also SDKs for Web3 app development, like [thirdweb](https://thirdweb.com/) or [Alchemy](https://www.alchemy.com/), which take away most of the complexity of blockchain development and provide a smooth developer experience.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+In Web3 apps, some data is stored on the blockchain, i.e., a decentralized database.
+How the data is stored is defined inside of smart contracts.
+However, it should be noted that onchain data on a public blockchain is visible to everyone and therefore not suitable for all types of data.
+Especially sensistive and private user data should not be stored on the blockchain, except it is encrypted.
```
```suggestion
Especially sensitive and private user data should not be stored on the blockchain, except if it is encrypted.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+It also provides a local testnet node for deploying and testing smart contracts, see [Anvil](https://book.getfoundry.sh/anvil/).
+- [Hardhat](https://hardhat.org): Hardhat is a development environment that facilitates smart contract development.
+It comes built-in with [Hardhat Network](https://hardhat.org/hardhat-network/docs/overview), a local Ethereum network node designed for development.
+Hardhat allows you to deploy your contracts, [run your tests](https://hardhat.org/tutorial/testing-contracts) and debug your code, all within the confines of your local machine.
```
```suggestion
Hardhat allows you to deploy your contracts, [run your tests](https://hardhat.org/tutorial/testing-contracts), and debug your code, all within the confines of your local machine.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+In Web2 apps, data is typically stored in a centralized database.
+In Web3 apps, some data is stored on the blockchain, i.e., a decentralized database.
+How the data is stored is defined inside of smart contracts.
+However, it should be noted that onchain data on a public blockchain is visible to everyone and therefore not suitable for all types of data.
```
```suggestion
However, it should be noted that onchain data on a public blockchain is visible to everyone and, therefore, not suitable for all types of data.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+
+### Testing and Deployment
+When it comes to testing and deployment of Web3 apps, there are some additional aspects to consider compared to Web2 apps.
+The intrinsic vulnerabilities of smart contracts, the immutability of blockchain data and interacting with a decentralized network call for specific testing and security measures.
```
```suggestion
The intrinsic vulnerabilities of smart contracts, the immutability of blockchain data, and the interactions with a decentralized network, call for specific testing and security measures.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+### Testing and Deployment
+When it comes to testing and deployment of Web3 apps, there are some additional aspects to consider compared to Web2 apps.
+The intrinsic vulnerabilities of smart contracts, the immutability of blockchain data and interacting with a decentralized network call for specific testing and security measures.
+Gladly, there are already many tools available to help developers ensure the security and reliability of their Web3 apps.
```
```suggestion
Gladly, many tools are already available to help developers ensure the security and reliability of their Web3 apps.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+
+#### Smart Contract Code Analysis
+To ensure the security of smart contracts, it is essential to analyze the code for vulnerabilities.
+There are several tools available that help developers to analyze their smart contracts for common security issues and vulnerabilities, for example:
```
```suggestion
There are several tools available that help developers analyze their smart contracts for common security issues and vulnerabilities, for example:
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+
+#### E2E Testing
+End-to-end (E2E) testing is a type of software testing that tests the complete flow of an application from start to finish.
+This specifically includes testing the frontend of your application, ensuring that the user interface works as expected.
```
```suggestion
This specifically includes testing the frontend of your application and ensuring that the user interface works as expected.
```

Tschakki replied:
```suggestion
This specifically includes testing the frontend of your application, which ensures that the user interface works as expected.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -171,14 +141,197 @@ In-app purchase fees can also be avoided because users no longer need to go thro
+
+#### Performance testing
+Performance testing is essential to ensure that your Web3 app can handle the expected load of users and their interactions.
+Measure and improve your apps performance, throughput, and resilience under different levels of traffic and transactions.
```
```suggestion
Measure and improve your app's performance, throughput, and resilience under different levels of traffic and transactions.
```

TalhaMaliktz commented on docs/building-on-lisk/web3-app-development.mdx:
```diff
@@ -25,7 +25,7 @@ import TabItem from '@theme/TabItem';
 On this page, you will learn what Web3 applications are, how they differ from Web 2.0 apps, and how to get started with Web3 app development on Lisk.
 
-## What are Web3 applications?
+## What are Web3 Applications?
```
I would wait for confirmation from marketing before choosing a capitalization style.

Tschakki replied:
I updated the headings now to be all sentence case 👍

TalhaMaliktz reviewed (changes requested):
Looking good already!

Tschakki committed:
Apply suggestions from code review

Co-authored-by: Muhammad Talha <13951043+TalhaMaliktz@users.noreply.github.com>

Tschakki committed:
Add link

Tschakki committed:
Merge branch 'liskdocs-19-mobile-dapp-guide' of github.com:LiskHQ/lisk-documentation into liskdocs-19-mobile-dapp-guide

Tschakki committed:
Merge branch 'development' into liskdocs-19-mobile-dapp-guide

Tschakki committed:
Use sentence case in all headings

Tschakki committed:
Merge branch 'development' into liskdocs-19-mobile-dapp-guide

TalhaMaliktz reviewed (approved):
LGTM! 👍🏼
